Market page

joaojvds opened the pull request:
-- Pagina da Market

joaojvds committed:
Iniciado desenvolvimento da pagina de market

-- Criação de uma versão inicial da pagina de market
-- Alterado valores de padding no arquivo layout, para que os arquivos das paginas tenham maior controle do layout
-- Adicionado botão para a pagina de market na navbar

joaojvds committed:
Designer do bloco de compra e venda

-- Terminado o designer inicial da parte de compra e venda

joaojvds committed:
Criado selects para alterar a moeda

-- Criado selects para alterar a moeda e um botão para trocar as moedas selecionadas

joaojvds committed:
Botão de troca

joaojvds committed:
Merge branch 'master' into market-page

GCastilho committed:
Corrigido coisa que deveria estar no master

-- Feito alguns tweaks no dev-db, criado volumes e arrumado alguns detalhes

Co-authred by @gabr1gus

GCastilho committed:
Implementado interface de update de preço

-- Implementado uma interface para os eventos de update de preço

joaojvds committed:
Novo designer da market

GCastilho committed:
Merge branch 'master' into market-http-api

GCastilho committed:
Método add não mais retorna 'SameCurrencyOperation'

-- O método add da MarketApi não mais faz um check próprio para saber se as currencies são diferentes, mas deixa essa validação para o mongoose

GCastilho committed:
Implementado o handler de inserção no orderbook

-- Implementado um entrypoint POST em /market/orderbook para ser usado para inserção de ordens
-- Implementado o esqueleto da suíte de testes do orderbook

GCastilho committed:
Criado um módulo para conter middlewares

-- Criado um módulo 'middlewares' para conter middlewares da API que são usados por vários entrypoints
-- Movido o middleware de autenticação para o módulo de middlewares
-- Adicionado um middleware de autenticação na orderbook

GCastilho committed:
Implementado testes para o POST ao orderbook

-- Implementado testes de autenticaçao e inserçao de ordens no entrypoint de POST no orderbook
-- Movido parsers da HTTP API da user para o index

GCastilho committed:
Implementado handler de DELETE de ordens

-- Implementado um handler DELETE em /v1/market/orderbook/:opid para executar remoções de ordens na MarketApi
-- Implementado testes desse handler

GCastilho committed:
Merge branch 'market-http-api' into market-page

GCastilho commented on .vscode/settings.json:
```diff
@@ -1,7 +1,6 @@
 {
 	"files.insertFinalNewline": true,
 	"editor.codeLens": true,
-	"editor.tabSize": 2,
```
Qual o motivo dessa remoção?

joaojvds committed:
Criado store market

-- Criado store market
-- Co-authored by @GCastilho and @gabr1gus

GCastilho committed:
Feito a pagina da market funcionar

-- Limpeza da casa, digo, pagina da market, e agora ela funciona. Yaaay
-- Adiçao de comentarios, agora da pra saber o que cada variavel (da market) faz!
-- API agora retorna NotEnoughFunds caso o usuario nao tenha saldo
-- Renomeado algumas variaveis devido a sanidade mental do programador
-- Adiçao de logs na market

Co-authrored by @gabr1gus

joaojvds committed:
Adicionado valor do total

-- Mudado currency code para upperCase
-- Adicionado conta para o price
-- Currency code agora não muda com buy e sell

joaojvds committed:
Adicionado handle para o evento "price_update"

-- Evento atualiza a store com o valor da market
-- Introduzida uma gambiarra que sera alterada provavelmente no próximo commit

joaojvds committed:
Metodo usando context

GCastilho committed:
Feito o botao de inverter currency inverter o preço

-- O botao de inverter corrency agora tambem inverte o preço mostrado no limitPrice
-- Corrigodo um bug onde o login nao estava dando erro ao dar erro de login

GCastilho committed:
Merge commit '8ac473302cc89a3618461619b4e24ce348c9abee' into market-page

-- Implementado interface de price_update

GCastilho committed:
Refatorado store da market

-- A store 'market' agora se chama orderbook, e sera um array de todas as ordens em aberto do cliente
-- O metodo 'orderbook' foi renomeado para 'add' e agora da lock do saldo
-- A interface MarketOrder foi movida do index da MarketApi para a interface 'market' das interfaces globais

GCastilho committed:
Atualizado interfaces do market-comunication

GCastilho committed:
Implementado listener para order_update

-- Implementado uma interface OrderUpdate para ser usada no evento de order_update da market
-- Implementado um listener na store do orderbook para o evento order_update, que atualiza a ordem no ordebook e o saldo do usuário

GCastilho committed:
Corrigido bug do nome da currency estar sendo setada para 0

-- Corrigido um bug onde, ao fazer a "limpeza" da trade, o nome da currency era setado para 0
-- Minor fixes

GCastilho committed:
Minor fixes

-- Corrigodo um bug no find de uma ordem na store

GCastilho committed:
Implementado basico do grafico de velas

-- Implementado um grafico de velas em d3 como um componente svelte
-- O zoom nao esta funcionando

joaojvds committed:
Refactored buySell

-- Alterado nomes de variáveis
-- Refatorado variáveis

gabr1gus committed:
work in progress

-- criado store e interface para priceHistory

gabr1gus committed:
Refatorado código do gráfico de velas

-- O código do gráfico de velas foi COMPLETAMENTE refeito e atualizado para a nova versão do d3 e os padrões mais recentes do javascript
-- O código agora é bonito, formoso, dá graça de olhar
-- O código agora usar os dados da store de preços, não de um CSV aleatório
-- O zoom agora está funcionando, yaaay
-- Co authored by @GCastilho, que tbm escreveu esse commit

gabr1gus committed:
pequena limpeza de codigo e alguns comentarios

-- Estava experimentando e tentando descobrir para que serve cada parte do código e resolvi mudar uns pequenos detalhes

gabr1gus committed:
Adicionado linhas para a régua vertical

-- Linhas que se adicionam e se removem com o zoom

gabr1gus committed:
pequenas melhorias no codigo da candle

-- Trocado id da div
-- Consertado bug na geraçao de linhas
-- Alterado tempo de transição para 600

gabr1gus committed:
depth graph

-- criaçao de um depth graph baseado no grafico de velas

gabr1gus committed:
depth graph

-- criaçao de um depth graph baseado no grafico de velas

GCastilho committed:
Renomeado componente de chart

-- Renomeado arquivo do componente de chart para candle
-- Removido lodash

GCastilho committed:
Corrigido erros do typescript

-- Corrigido erros do typescript no componente de Candle

GCastilho committed:
Corrigido bug que impedia um grafico vazio

-- Corrigido um bug que impedia a renderizaçao do grafico de candles quando nao havia nenhum dado
-- O array de prices agora e informado atraves de uma propriedade exportada
Co-authored by @gabr1gus

gabr1gus committed:
stage para trocar de pc

-- provavelmente vou remover esse commit

gabr1gus committed:
Revert "stage para trocar de pc"

This reverts commit 5860ad4955158487b3639b66d8ae00c30b3ab2f7.

joaojvds committed:
Totem removido

-- Ajustado tamanho da área de compra e venda.
-- Cores da pagina de market ajustadas

gabr1gus committed:
Adicionado Scroll para o grafico de candles

-- esta meio bugado
-- esta muito engraçado
-- lembrar gabriel do futuro que tem que mudar a inicializaçao dos dados do update

gabr1gus committed:
corrigido parcialmente o update do grafico

-- AGORA O GRAFICO NAO PARECE UM CORAÇAO PULSANTE
-- quebra facilmente
-- adicionado mais coisa pra ser refatorado

gabr1gus committed:
Corrigido escala e modificaçoes do grafico

-- Adicionado stems no update
-- Agora update esta funcionando de maneira semi estavel
-- Alguns problemas em transiçao, mas problemas aceitaveis
-- Need refactoring
-- NEED URGENT REFACTORING
-- probably heavy on the gpu/prossessor front
-- my gpu can handle it

gabr1gus committed:
Refactorado um pouco

-- adicionado types
-- removido algumas declaraçoes desnecessarias
-- muita coisa pra fazer ainda

gabr1gus committed:
um pouco mais de refactor

-- adicionado comentarios
-- ainda falta adicionar comentarios
-- melhorado alguns problemas com o gráfico

joaojvds committed:
Criação da store orders

-- Criado um arquivo orders.ts
-- Colocado comentários na pagina de transactions

gabr1gus committed:
corrigido scroll permanente

-- agora o scroll so funciona se estiver no canto direito

joaojvds committed:
Criado entrypoint na HTTPAPI

-- Implementado entrypoint para pegar as ordens em aberto
-- Implementado função de fetch na store orderbook

gabr1gus committed:
Grafico de depth refatorado

-- Alguns problemas resolvidos
-- Agora o Ymin é ancorado em 0

gabr1gus committed:
Merge remote-tracking branch 'origin/depth-graph' into depth-graph

joaojvds committed:
Implementado lista de ordens abertas na market page

-- A lista de ordens abertas aparecem na market
-- Apagado antiga pagina de orders

joaojvds committed:
Adicionado abas na pagina de market

-- Implementado abas para as lista de ordens abertas e fechadas

gabr1gus committed:
adicionado store para depth

-- agora os dados do marketDepth vem da store
-- agora ele atualiza corretamente
-- pequenas mudanças no nome de variaveis

gabr1gus committed:
Pequeno refactor com o intuito de corrigir um bug

-- o bug aparentemente foi corrigidog
-- eu nao sei como o bug acontecia
-- esse grafico esta amaldiçoado

GCastilho committed:
Merge branch 'master' into market-page

joaojvds committed:
Criado store trades

-- Criado store para o armazenamento das ordens fechadas
-- Alterado as posições dos itens da pagina de market

GCastilho committed:
Criado interface da ordem de Trade

-- Criado interface da ordem de uma trade já executada que será retornado ao usuário

GCastilho committed:
Merge branch 'ordens-page' into market-page

gabr1gus committed:
Merge branch 'depth-graph' into market-page

-- Corrigido conflito com arquivos incompatíveis
-- corrigido erro de map incorreto na depth
-- svelte-tabs colocada como dev-dependency

GCastilho committed:
Adicionado query customizada para buscar trades

-- Adicionado query para a Trade que procura e formata uma trade referente a um usuário
-- Adicionado teste para a nova query

joaojvds committed:
Alterado informações na lista de ordens

-- Agora a lista de ordens em aberto mostra os dados no formato certo para o usuário 
-- Alterado tamanho das tabelas
-- Alterado propriedade recebida pelo get para se adequar a interface

GCastilho committed:
Corrigido um bug que fazia a ordem ser enviada com type errado

-- Corrigido um bug onde uma ordem aberta era enviada como um documento do mongo, não como uma MarketOrder
-- Interface MarketOrder foi renomeada para OrderRequest, a MarketOrder agora extende a MarketOrder e tem um opid
-- O método toJSON do documento de uma ordem agora retorna um documento no formato MarketOrder

GCastilho committed:
Alterado interface de UserTrade

-- Alterado a interface de UserTrade para também enviar o opid e as currencies daquela trade

GCastilho committed:
Adicionado rota para retornar trades

-- Adicionado uma rota /trades que retorna um array de trades ao ser chamada com um GET

GCastilho committed:
Merge branch 'closed-orders-entrypoint' into market-page

GCastilho committed:
Corrigido interfaces erradas na store de Trade

-- Corrigido store de Trade estar usando interface de Ordem ao invés da interface de MarketTrade

GCastilho committed:
Implementado evento de new_trade

-- Implementado um evento (supostamente temporário) de uma nova trade, emitido quando uma nova trade é completada
-- Implementado um handler na store de trades que irá adicionar uma nova ordem ao início da store quando o evento de new_trade for recebido

GCastilho committed:
Adicionado uma interfaces numas paradas

-- Adicionado interfaces de MarketOrder e MarketTrade nos componentes de tableRow das ordens

joaojvds committed:
Mudado *SÓ* uma coisa

-- acrescentado o timestamp  na interface

joaojvds committed:
Ajustado items das listas de ordens

-- Os dados das listas agora seguem o padrão das interfaces
-- Os valores das moedas agora tem o seu numero de casas setado de acordo com a moeda e o código de sua respectiva moeda

joaojvds committed:
Implementado fetch na store depth

-- Implementado método para fazer o fetch dos dados para o gráfico

joaojvds committed:
Adicionado um listener para o evento do socket

-- Ainda não tenho certeza se esta funcionando, preciso do nb para confirmar

joaojvds committed:
Implementado função fetch na store prices

-- Função fetch é executada quando houver alteração das currencies
-- Colocado um listening para o evento de atualização do gráfico

GCastilho committed:
Adicionado rota para retornar o historico de preços

-- Adicioando uma rota em /market/candle para retornar o historico de preços do sistema

joaojvds committed:
Criado store para o preço da market

-- Criado store para o preço da market no arquivo _buysell
-- Agora ele enviar por ordem alfabética

joaojvds committed:
Consertado os gráficos 

-- Arrumado o erro do gráfico ao pegar dados da store
-- O preço da market agora aparece na pagina

joaojvds committed:
Merge branch 'master' into market-page

GCastilho committed:
Implementado evento de price_history

-- Collection de prices agora emite o evento price_history na call da createOne
-- O evento de price_history é emitido pelo socket do servidor
-- Implementado update do gráfico de candle ao receber o evento de price_history

GCastilho committed:
Merge branches 'market-page' and 'market-page' of github.com:GCastilho/unip-tcc into market-page

GCastilho committed:
Bugfix

-- Corrigido um bug onde eu disse pro compilador confiar em mim e eu estava errado

GCastilho reviewed (approved):
Vai q a apresentação é amanhã, tá td funfando sim
